src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/util/OcrFormValidationHelper.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.util;
+
+import org.apache.commons.validator.routines.EmailValidator;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrDataField;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+public final class OcrFormValidationHelper {
+
+    private OcrFormValidationHelper() {
+        // util class
+    }
+
+    public static List<String> getOcrFieldNames(List<OcrDataField> ocrData) {
+        return ocrData
+            .stream()
+            .map(field -> field.key)
+            .collect(toList());
+    }
+
+    public static List<String> findBlankFields(List<String> fieldNames, List<OcrDataField> ocrData) {
+        return fieldNames
+            .stream()
+            .filter(field -> isBlank(OcrFormValidationHelper.findOcrFormFieldValue(field, ocrData)))
+            .collect(toList());
+    }
+
+    public static List<String> findMissingFields(List<String> optionalFields, List<String> ocrInputFields) {
+        return optionalFields.stream()
+            .filter(item -> !ocrInputFields.contains(item))
+            .collect(Collectors.toList());
+    }
+
+    public static String findOcrFormFieldValue(String fieldName, List<OcrDataField> ocrData) {
```

Comment:
Perhaps not for this PR, but at some point we might also verify if a field is not duplicated.

Reply:
Added duplicate keys validation